src/caffe/proto/caffe.proto

```diff
@@ -347,6 +347,9 @@ message LayerParameter {
   // Parameters shared by loss layers.
   optional LossParameter loss_param = 101;
 
+  // Specify name of source class
+  optional string source_class = 143;
```

Comment:
This should be a field in `MemoryDataParameter`, not `LayerParameter` if I understand this patch.